docs/reference/sync-configuration.md

```diff
@@ -27,20 +27,26 @@ The flagd accepts a string argument, which should be a JSON representation of an
-
-The `uri` field values **do not** follow the [URI patterns](#uri-patterns). The provider type is instead derived
-from the `provider` field. Only exception is the remote provider where `http(s)://` is expected by default. Incorrect
-URIs will result in a flagd start-up failure with errors from the respective sync provider implementation.
+| Field       | Type               | Note                                                                                                                                                                          |
+| ----------- | ------------------ | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| uri         | required `string`  | Flag configuration source of the sync                                                                                                                                         |
+| provider    | required `string`  | Provider type - `file`, `kubernetes`, `http`, or `grpc`                                                                                                                       |
+| authHeader  | optional `string`  | Used for http sync; set this to include the complete `Authorization` header value for any authentication scheme (e.g., "Bearer token_here", "Basic base64_credentials", etc.) |
+| bearerToken | optional `string`  | (Deprecated) Used for http sync; token gets appended to `Authorization` header with [bearer schema](https://www.rfc-editor.org/rfc/rfc6750#section-2.1)                       |
+| interval    | optional `uint32`  | Used for http sync; requests will be made at this interval. Defaults to 5 seconds.                                                                                            |
+| tls         | optional `boolean` | Enable/Disable secure TLS connectivity. Currently used only by gRPC sync. Default (ex: if unset) is false, which will use an insecure connection                              |
+| providerID  | optional `string`  | Value binds to grpc connection's providerID field. gRPC server implementations may use this to identify connecting flagd instance                                             |
+| selector    | optional `string`  | Value binds to grpc connection's selector field. gRPC server implementations may use this to filter flag configurations                                                       |
+| certPath    | optional `string`  | Used for grpcs sync when TLS certificate is needed. If not provided, system certificates will be used for TLS connection                                                      |
+
+### Important Notes
+
+- **Mutual Exclusivity**: `authHeader` and `bearerToken` cannot be used simultaneously. Setting both fields in the configuration
+  results in an error, preventing the application from starting.
```

Comment:
Nit: this is fine, but I think another option would do just mention in each they are incompatible with each other, respectively, for example:

`Used for http sync; set this to include the complete `Authorization` header value for any authentication scheme (e.g., "Bearer token_here", "Basic base64_credentials", etc.). Cannot be used with `bearerToken`.

Reply:
Added